impeller/entity/contents/text_contents.cc

```diff
@@ -147,23 +156,29 @@ static bool CommonRender(
         return false;
       }
 
-      auto atlas_position = atlas_glyph_pos->origin;
-      auto atlas_glyph_size =
-          Point{atlas_glyph_pos->size.width, atlas_glyph_pos->size.height};
       auto offset_glyph_position =
           glyph_position.position + glyph_position.glyph.bounds.origin;
 
+      auto uv_scaler_a = atlas_glyph_pos->size / atlas_size;
+      auto uv_scaler_b = (Point::Round(atlas_glyph_pos->origin) / atlas_size);
+      auto translation =
+          Matrix(1, 0, 0, 0,  //
+                 0, 1, 0, 0,  //
+                 0, 0, 1, 0,  //
+                 offset_glyph_position.x, offset_glyph_position.y, 0, 1);
+
       for (const auto& point : unit_points) {
         typename VS::PerVertexData vtx;
-        vtx.unit_position = point;
-        vtx.destination_position = offset_glyph_position;
-        vtx.destination_size = Point(glyph_position.glyph.bounds.size);
-        vtx.source_position = atlas_position;
-        vtx.source_glyph_size = atlas_glyph_size;
+        auto position = PositionForGlyphPosition(
+            translation, point, glyph_position.glyph.bounds.size);
+        vtx.uv = point * uv_scaler_a + uv_scaler_b;
+        vtx.position = position;
```

Comment:
I like that we're not passing as much needlessly interpolated data to the fragment shader anymore (🤢), but I wonder if the 16 byte savings per vertex is worth the extra per character CPU work. My guess is that it's worth it.

Reply:
you also have to pay the CPU cost of copying that extra data too. Its probably slightly more expensive on the raster side of things but hard to tell exactly.